jmespath/parser.py

```diff
@@ -371,8 +379,10 @@ def _error_led_token(self, token):
                                     token['type'], 'Invalid token')
 
     def _match(self, token_type=None):
-        if self._current_token() == token_type:
-            self._advance()
+        # inline'd self._current_token()
```

Comment:
What's the advantage of inlining here? It's not clear whether the performance enhancements are due to this or from moving away from a binary tree node. Also, does inlining a function in Python really make much of a difference?